openmc/tallies.py

```diff
@@ -1634,83 +1638,33 @@ def _align_tally_data(self, other):
 
             # Determine the factors by which each tally operands' data arrays
             # must be tiled or repeated for the tally outer product
-            other_tile_factor = 1
-            self_repeat_factor = 1
             for filter in diff1:
-                other_tile_factor *= filter.num_bins
+                tile_factors[0] *= filter.num_bins
             for filter in diff2:
-                self_repeat_factor *= filter.num_bins
-
-            # Tile / repeat the tally data for the tally outer product
-            self_shape = list(self_mean.shape)
-            other_shape = list(other_mean.shape)
-            self_shape[0] *= self_repeat_factor
-            self_mean = np.repeat(self_mean, self_repeat_factor)
-            self_std_dev = np.repeat(self_std_dev, self_repeat_factor)
-
-            if self_repeat_factor == 1:
-                other_shape[0] *= other_tile_factor
-                other_mean = np.repeat(other_mean, other_tile_factor, axis=0)
-                other_std_dev = np.repeat(other_std_dev, other_tile_factor,
-                                          axis=0)
-            else:
-                other_mean = np.tile(other_mean, (other_tile_factor, 1, 1))
-                other_std_dev = np.tile(other_std_dev, (other_tile_factor, 1, 1))
-
-            # NumPy repeat and tile routines return 1D flattened arrays
-            # Reshape arrays as 3D with filters, nuclides and scores axes
-            self_mean.shape = tuple(self_shape)
-            self_std_dev.shape = tuple(self_shape)
-            other_mean.shape = tuple(other_shape)
-            other_std_dev.shape = tuple(other_shape)
+                repeat_factors[0] *= filter.num_bins
 
         if self.nuclides != other.nuclides:
 
             # Determine the number of paired combinations of nuclides
             # between the two tallies and repeat arrays along nuclide axes
-            self_repeat_factor = other.num_nuclides
```

Comment:
Ditto - move the comment above the `if` and remove the newline to tighten up this (now) tiny code block